src/tiled/mapeditor.cpp

```diff
@@ -386,6 +392,9 @@ void MapEditor::setCurrentDocument(Document *document)
             mZoomable->setComboBox(mZoomComboBox);
         }
 
+        connect(mapView, &MapView::focused,
+                this, [this, mapDocument](){ mPropertiesDock->setDocument(mapDocument); });
```

Comment:
Alright, I missed that. I added the currentObjectChanged, it works fine, but it won't work if the clicked object is the same, as the signal won't be emitted.

So I can connect all docks' focus signals to trigger setDocument which might not be that good.